airflow/providers/amazon/aws/example_dags/example_hive_to_dynamodb.py

```diff
@@ -0,0 +1,137 @@
+    client.create_table(
+        TableName=DYNAMODB_TABLE_NAME,
+        KeySchema=[
+            {'AttributeName': DYNAMODB_TABLE_HASH_KEY, 'KeyType': 'HASH'},
+        ],
+        AttributeDefinitions=[
+            {'AttributeName': DYNAMODB_TABLE_HASH_KEY, 'AttributeType': 'N'},
+        ],
+        ProvisionedThroughput={'ReadCapacityUnits': 20, 'WriteCapacityUnits': 20},
+    )
+
+    # DynamoDB table creation is nearly, but not quite, instantaneous.
+    # Wait for the table to be active to avoid race conditions writing to it.
+    waiter = client.get_waiter('table_exists')
+    waiter.wait(TableName=DYNAMODB_TABLE_NAME, WaiterConfig={'Delay': 1})
+
+
+@task
+def get_dynamodb_item_count():
+    """
+    A DynamoDB table has an ItemCount value, but it is only updated every six hours.
+    To verify this DAG worked, we will scan the table and count the items manually.
+    """
+    table = DynamoDBHook(resource_type='dynamodb').conn.Table(DYNAMODB_TABLE_NAME)
+
+    response = table.scan(Select='COUNT')
+    item_count = response['Count']
+
+    while 'LastEvaluatedKey' in response:
+        response = table.scan(Select='COUNT', ExclusiveStartKey=response['LastEvaluatedKey'])
+        item_count += response['Count']
+
+    print(f'DynamoDB table contains {item_count} items.')
+
+
+# Included for sample purposes only; in production you wouldn't delete your fresh backup.
+def delete_dynamodb_table_fn():
+    DynamoDBHook(client_type='dynamodb').conn.delete_table(TableName=DYNAMODB_TABLE_NAME)
+
+
+# Included for sample purposes only; in production this should
+# be configured in the environment and not be part of the DAG.
+# Note: The 'hiveserver2_default' connection will not work if Hive
+# is hosted on EMR.  You must set the host name of the connection
+# to match your EMR cluster's hostname.
+@task
+def configure_hive_connection():
+    db.merge_conn(
+        Connection(
+            conn_id=HIVE_CONNECTION_ID,
+            conn_type='hiveserver2',
+            host=HIVE_HOSTNAME,
+            port=10000,
+        )
+    )
+
+
+with DAG(
+    dag_id='example_hive_to_dynamodb',
+    schedule_interval=None,
+    start_date=datetime(2021, 1, 1),
+    tags=['example'],
+    catchup=False,
+) as dag:
+    # Add the prerequisites docstring to the DAG in the UI.
+    dag.doc_md = __doc__
+
+    # [START howto_transfer_hive_to_dynamodb]
+    backup_to_dynamodb = HiveToDynamoDBOperator(
+        task_id='backup_to_dynamodb',
+        hiveserver2_conn_id=HIVE_CONNECTION_ID,
+        sql=HIVE_SQL,
+        table_name=DYNAMODB_TABLE_NAME,
+        table_keys=[DYNAMODB_TABLE_HASH_KEY],
+    )
+    # [END howto_transfer_hive_to_dynamodb]
+
+    # Using 'all_done' so even if an intermediate step fails, the DAG will clean up after itself.
+    delete_dynamodb_table = PythonOperator(
+        task_id="delete_dynamodb_table",
+        python_callable=delete_dynamodb_table_fn,
+        trigger_rule='all_done',
+    )
```

Comment:
Should convert this to TaskFlow API as well.

Reply:
Wasn't that because using `chain()`with TaskFlow API was not compatible with Airflow 2.1? You should be able to pass a `trigger_rule` directly to the decorator like you're trying to do with Airflow 2.1+

Reply:
Alright, I'll revert that and see.   Maybe I just misread the error message.    I'm, in a meeting, but should have it pushed within the hour

Reply:
That didn't age well.  :stuck_out_tongue:   Rebased and pushed reversion.